apps/pkeyutl.c

```diff
@@ -419,8 +419,13 @@ static EVP_PKEY_CTX *init_ctx(int *pkeysize,
     if (!pkey)
         goto end;
 
-    ctx = EVP_PKEY_CTX_new(pkey, e);
-
+    if ((keyform == FORMAT_ENGINE) && (strncmp(ENGINE_get_name(e),"pkcs11 engine", strlen("pkcs11 engine"))==0)) {
+      fprintf(stderr, "engine name = \"%s\"\n", ENGINE_get_name(e));
+      ctx = EVP_PKEY_CTX_new(pkey, NULL);
+    } else {
+      ctx = EVP_PKEY_CTX_new(pkey, e);
+    }
```

Comment:
This change makes no sense to me.  If `keyform` is `FORMAT_ENGINE`, `pkey` will be associated with the engine (`e`), and according to my reading of `EVP_PKEY_CTX_new` (*), it will ignore the engine parameter and use the one from `pkey` instead.  So unless I'm reading our code wrong, having `NULL` instead of `e` there doesn't matter at all.